packages/console/src/Components/Setup/Setup.tsx

```diff
@@ -69,38 +85,69 @@ function Setup() {
   useEffect(() => {
     if (steps.length === 1) {
       navigate("/console");
-    } else if (steps.length === 2 && steps[1] === StepSignUp) {
+    } else if (steps.length === 2 && steps[1] === "create-account") {
       setReadyToRender(true);
       setStep(1);
     } else {
       setReadyToRender(true);
       setDocumentTitle("Setup");
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  }, [steps, navigate]);
 
-  /**
-   */
   useEffect(() => {
-    if (!SetupComponent) {
+    if (!currentStep) {
       navigate("/console");
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [SetupComponent]);
+  }, [currentStep, navigate]);
 
-  if (!readyToRender || !SetupComponent) {
+  if (!readyToRender || !currentStep) {
     return null;
   }
 
+  if (currentStep === "master-password") {
```

Comment:
Broken down into separate `if` statements for each step, makes the code "taller" but easier to read and debug